sensor/kubernetes/listener/resources/deployment_store.go

```diff
@@ -167,10 +168,38 @@ func (ds *DeploymentStore) FindDeploymentIDsByLabels(namespace string, sel selec
 	return
 }
 
+func (ds *DeploymentStore) findDeploymentIDsByImageNoLock(image *storage.Image) set.Set[string] {
+	ids := set.NewStringSet()
+	for _, d := range ds.deployments {
+		for _, c := range d.GetContainers() {
+			if imagecacheutils.CompareImageCacheKey(c.GetImage(), image) {
+				ids.Add(d.GetId())
+				// The deployment id is already the set, we can break here
+				break
+			}
+		}
+	}
+	return ids
+}
+
+// FindDeploymentIDsByImages returns a slice of deployment ids based on matching images
+func (ds *DeploymentStore) FindDeploymentIDsByImages(images []*storage.Image) []string {
```

Comment:
Maybe we could add some tests in deployment store for this function?

Reply:
I've added some tests.